gnocchi/rest/aggregates/processor.py

```diff
@@ -24,28 +24,24 @@
 from gnocchi import carbonara
 from gnocchi.rest.aggregates import operations as agg_operations
 from gnocchi import storage as gnocchi_storage
+from gnocchi import utils
 
 
 LOG = daiquiri.getLogger(__name__)
 
 
 class UnAggregableTimeseries(Exception):
     """Error raised when timeseries cannot be aggregated."""
-    def __init__(self, reason):
+    def __init__(self, references, reason):
+        self.references = references
         self.reason = reason
         super(UnAggregableTimeseries, self).__init__(reason)
 
-
-class MetricUnaggregatable(Exception):
-    """Error raised when metrics can't be aggregated."""
-
-    def __init__(self, metrics_and_aggregations, reason):
-        self.metrics_and_aggregations = metrics_and_aggregations
-        self.reason = reason
-        metrics = ("%s/%s" % (m.id, a) for (m, a) in metrics_and_aggregations)
-        super(MetricUnaggregatable, self).__init__(
-            "Metrics %s can't be aggregated: %s" % (
-                ", ".join(metrics), reason))
+    def jsonify(self):
+        return {
+            "cause": "Metrics can't being aggregated: %s" % self.reason,
+            "detail": self.references
```

Comment:
add a "reason" field with `self.reason` directly rather than writing it in `cause`